### PR TITLE
fix: links and a couple typos & clarifications.

### PR DESCRIPTION
**Pull request by lovettchris**

### CODE REVIEW

File: lean-toolchain [hunk 1]
```diff
@@ -0,0 +1 @@
+leanprover/lean4:nightly
```
Comment (Kha) [inline lean-toolchain:1]: There are no .lean files in this repo anyway, are there?

File: structures_and_records.md [hunk 2]
```diff
@@ -49,7 +49,8 @@ structure Point (α : Type u) where
 Values of type ``Point`` are created using ``Point.mk a b``, and the
 fields of a point ``p`` are accessed using ``Point.x p`` and
 ``Point.y p``. The structure command also generates useful recursors and
-theorems. Here are some of the constructions generated for the
+theorems. Dot notation is also supported so you can write `p.x`
```
Comment (Kha) [inline structures_and_records.md:52]: Duplicate of line 95ff

**Review by Kha — CHANGES_REQUESTED** [review]

File: lean-toolchain (hunk 1 above)
Comment (lovettchris) [reply lean-toolchain:1]: Oh, it is for `mdbook test`.

Comment (Kha) [reply lean-toolchain:1]: Let's use a pinned release then and bump it when we fix breakage from upstream. Tests should be deterministic.

Comment (lovettchris) [reply lean-toolchain]: Ok, I pinned it to leanprover/lean4:nightly-2022-02-01

File: structures_and_records.md (hunk 2 above)
Comment (lovettchris) [reply structures_and_records.md]: True, and good point, but I felt uneasy with ``Point.x p`` and wanted to be rescued by the dot notation sooner :-)

**Review by Kha — APPROVED** [review]

**Pull request merged**